### PR TITLE
UI financing statement step2 add reg party.

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#7136

*Description of changes:*
-- Add registering party from auth to step 2 of create financing statement.
-- Partial completion of auth call: waiting for address change (7140).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/parties/RegisteringParty.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    var parties: AddPartiesIF = getAddSecuredPartiesAndDebtors.value
+
+    /** First time get read only registering party from the auth api. After that get from the store. */
+    onMounted(async () => {
+      if (parties.registeringParty === null) {
+        try {
+          const regParty = await getRegisteringPartyFromAuth()
+          parties.registeringParty = regParty
+          setAddSecuredPartiesAndDebtors(parties)
+          console.log('RegisteringParty.vue onMounted regParty=' + JSON.stringify(regParty))
```
Comment (jordiwes) [inline ppr-ui/src/components/parties/RegisteringParty.vue]: remove this?

File: ppr-ui/src/resources/tableHeaders.ts [hunk 2]
```diff
@@ -260,3 +260,36 @@ export const partyTableHeaders: Array<BaseHeaderIF> = [
...
+    text: 'Email Address',
+    value: 'emailAddress'
+  },
+  {
+    class: 'column-mds',
+    sortable: false,
+    text: 'Registering Party Code',
+    value: 'code'
+  },
+  {
```
Comment (jordiwes) [inline ppr-ui/src/resources/tableHeaders.ts:289]: Probably don't need this edit column

File: ppr-ui/src/utils/auth-helper.ts [hunk 3]
```diff
@@ -31,3 +38,45 @@ export function getKeycloakRoles (): Array<string> {
...
+        if (!data) {
+          throw new Error('Unable to obtain Registering Party from Account Information.')
+        }
+        // Remove null defaults when api address available.
+        const address: AddressIF = {
+          street: data?.contact?.street || 'NA',
+          streetAdditional: data?.contact?.streetAdditional,
+          city: data?.contact?.city || 'NA',
+          region: data?.contact?.region || 'BC',
+          postalCode: data?.contact?.postalCode || 'V8R1V1',
```
Comment (jordiwes) [inline ppr-ui/src/utils/auth-helper.ts:63]: Are we defaulting the postal code to this? ;)

File: ppr-ui/src/utils/auth-helper.ts [hunk 4]
```diff
@@ -31,3 +38,45 @@ export function getKeycloakRoles (): Array<string> {
...
+          country: data?.contact?.country || 'CA',
+          deliveryInstructions: ''
+        }
+        const party: PartyIF = {
+          businessName: data.name,
+          personName: { first: 'Test', last: 'Person' },
```
Comment (jordiwes) [inline ppr-ui/src/utils/auth-helper.ts:69]: probably want to remove Test Person?

File: ppr-ui/tests/unit/RegisteringParty.spec.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    expect(rowCount).toEqual(1)
+  })
+
+  it('displays the correct data in the table rows', () => {
+    const item1 = wrapper.vm.$el.querySelectorAll('.v-data-table .registering-row')[0]
+
+    expect(item1.querySelectorAll('td')[0].textContent).toContain('ABC REGISTERING')
+    expect(item1.querySelectorAll('td')[1].textContent).toContain('1234 Fort St.')
+  })
+})
```
Comment (jordiwes) [inline ppr-ui/tests/unit/RegisteringParty.spec.ts:84]: Perhaps add a few lines to the PartySummary test?

File: ppr-ui/src/components/parties/RegisteringParty.vue (hunk 1 above)
Comment (doug-lovett) [reply ppr-ui/src/components/parties/RegisteringParty.vue]: Will remove.

File: ppr-ui/src/resources/tableHeaders.ts (hunk 2 above)
Comment (doug-lovett) [reply ppr-ui/src/resources/tableHeaders.ts:289]: Leaving for now: ui may change to allow edits on RP.

File: ppr-ui/src/utils/auth-helper.ts (hunk 3 above)
Comment (doug-lovett) [reply ppr-ui/src/utils/auth-helper.ts:63]: Comment in code: // Remove null defaults when api address available.
Defaults are temporary to display address info and pass validation. All address defaults will be removed once the auth api call returns address info.

File: ppr-ui/src/utils/auth-helper.ts (hunk 4 above)
Comment (doug-lovett) [reply ppr-ui/src/utils/auth-helper.ts:69]: Comment in code: // Remove null defaults when api address available.
Defaults are temporary to display address info and pass validation. All address defaults will be removed once the auth api call returns address info.

File: ppr-ui/tests/unit/RegisteringParty.spec.ts (hunk 5 above)
Comment (doug-lovett) [reply ppr-ui/tests/unit/RegisteringParty.spec.ts:84]: Will do.

**Review by jordiwes — APPROVED** [review]
LGTM

**Pull request merged**